app/src/main/java/com/ddoczi/tasky/authentication/presentation/login/LoginState.kt

```diff
@@ -0,0 +1,11 @@
+package com.ddoczi.tasky.authentication.presentation.login
+
+data class LoginState(
+    val email: String = "",
+    val password: String = "",
+    val isEmailValid: Boolean = false,
+    val isPasswordValid: Boolean = false,
+    val emailError: Boolean = false,
```

Comment:
`emailError` would let me think this is a String. Usually, I'd prefer calling booleans so that they are answerable with yes or no. How about `isEmailError` instead? Or you actually make it a nullable string, so you can also pass the error message right away. And if there's no error, you keep it at null